aggregator/internal/pkg/aggregator.go

```diff
@@ -377,3 +385,51 @@ func (agg *Aggregator) AddNewTask(batchMerkleRoot [32]byte, senderAddress [20]by
 	agg.AggregatorConfig.BaseConfig.Logger.Info("- Unlocked Resources: Adding new task")
 	agg.logger.Info("New task added", "batchIndex", batchIndex, "batchIdentifierHash", "0x"+hex.EncodeToString(batchIdentifierHash[:]))
 }
+
+// Long-lived goroutine that periodically checks and removes old Tasks from stored Maps
+// It runs every period and removes all tasks older than blocksOld
+// This was added because each task occupies memory in the maps, and we need to free it to avoid a memory leak
+func (agg *Aggregator) ClearTasksFromMaps(period time.Duration, blocksOld uint64) {
+	defer func() {
+		err := recover() //stops panics
+		if err != nil {
+			agg.logger.Error(err.(string))
```

Comment:
Casting to string may fail, but `error` has an `Error()` method to format as string:
```suggestion
			agg.logger.Error(err.Error())
```